pkg/databasemonitoring/aws/aurora.go

```diff
@@ -0,0 +1,116 @@
+package aws
+
+import (
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"hash/fnv"
+	"regexp"
+	"strconv"
+
+	"strings"
+)
+
+const regexPattern = `^([a-z]+-[a-z]+-\d+)[a-z]$`
+
+var awsRegionRegex = regexp.MustCompile(regexPattern)
+
+// AuroraCluster represents an Aurora cluster
+type AuroraCluster struct {
+	Instances []*Instance `json:"instances,omitempty"`
+}
+
+// Instance represents an Aurora instance
+type Instance struct {
+	Endpoint   string `json:"endpoint,omitempty"`
+	Port       int64  `json:"port,omitempty"`
+	Region     string `json:"region,omitempty"`
+	IamEnabled bool   `json:"iam_enabled,omitempty"`
+}
+
+// GetAuroraClusterEndpoints queries an AWS account for the endpoints of an Aurora cluster
+// requires the dbClusterIdentifier for the cluster
+func (c *Client) GetAuroraClusterEndpoints(dbClusterIdentifiers []string) (map[string]*AuroraCluster, error) {
+	if len(dbClusterIdentifiers) == 0 {
+		return nil, fmt.Errorf("at least one database cluster identifier is required")
+	}
+	idVals := make([]*string, len(dbClusterIdentifiers))
+	for i, id := range dbClusterIdentifiers {
+		idVals[i] = aws.String(id)
+	}
+	clusterInstances, err := c.client.DescribeDBInstances(
+		&rds.DescribeDBInstancesInput{
+			Filters: []*rds.Filter{
+				{
+					Name:   aws.String("db-cluster-id"),
+					Values: idVals,
+				},
+			},
+		})
+	if err != nil {
+		return nil, fmt.Errorf("error describing aurora DB clusters: %v", err)
+	}
+	clusters := make(map[string]*AuroraCluster, 0)
+	for _, db := range clusterInstances.DBInstances {
+		if db.Endpoint != nil && db.DBClusterIdentifier != nil {
```

Comment:
once we get the db instances, i think it's safer to do one additional filtering on `DBInstanceStatus` to only get available instances.
https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/accessing-monitoring.html#Overview.DBInstance.Status

Reply:
makes sense! should we only return `Available` or are there other statues on there where the db instance could still receive requests? Some of these `Configuring-` statues don't make it clear if the database could still take traffic or not.

Reply:
Good question. ideally we could return instances that are "connectable" but hard to tell from those status. I think it's safer to only return available instances to start with

Reply:
kk, updated to do that